process/smartContract/process.go

```diff
@@ -1246,24 +1263,50 @@ func (sc *scProcessor) processSimpleSCR(
 	scResult *smartContractResult.SmartContractResult,
 	dstAcc state.UserAccountHandler,
 ) error {
-	outputAccount := &vmcommon.OutputAccount{
-		Code:         scResult.Code,
-		CodeMetadata: scResult.CodeMetadata,
-		Address:      scResult.RcvAddr,
+	if scResult.Value.Cmp(zero) <= 0 {
+		return nil
 	}
 
-	sc.updateSmartContractCode(dstAcc, outputAccount, scResult)
+	isPayable, err := sc.IsPayable(scResult.RcvAddr)
+	if err != nil {
+		return err
+	}
+	if !isPayable && !bytes.Equal(scResult.RcvAddr, scResult.OriginalSender) {
```

Comment:
So we allow transfer from the smart contract to itself, even if it's not payable?

Reply:
yes. There is a case when sc A calls a sc B which is not payable - and the scr returns here -> sc A should receive the money